Prep/Source/Alignment/ICPRigid3DImageRegistration/source/MatrixRigid3DImageRegistration.cxx

```diff
@@ -0,0 +1,606 @@
+    parser.add_option("--visualizeResult").action("store").type("bool").set_default(false).help("A flag to visualize the registration result.");
+    parser.add_option("--solutionSegmentation").action("store").type("string").set_default("").help("The filename of the aligned segmentation of source image.");
+    parser.add_option("--solutionRaw").action("store").type("string").set_default("").help("The filename of the aligned raw source image.");
+    parser.add_option("--sourceRaw").action("store").type("string").set_default("").help("The raw source image.");
+    parser.add_option("--solutionTransformation").action("store").type("string").set_default("").help("The filename of the textfile containing the transformation matrix.");
+    */
+    parser.add_option("--targetDistanceMap").action("store").type("string").set_default("").help("The distance map of target image.");
+    parser.add_option("--sourceRaw").action("store").type("string").set_default("").help("The raw source image.");
+    parser.add_option("--solutionRaw").action("store").type("string").set_default("").help("The filename of the aligned raw source image.");
+    //parser.add_option("--solutionTransformation").action("store").type("float").set_default("").help("The filename of the textfile containing the transformation matrix.");
+    parser.add_option("--m_00").action("store").type("float").set_default("").help("(0,0) element of the tranformation matrix.");
+    parser.add_option("--m_01").action("store").type("float").set_default("").help("(0,1) element of the tranformation matrix.");
+    parser.add_option("--m_02").action("store").type("float").set_default("").help("(0,2) element of the tranformation matrix.");
+    parser.add_option("--m_10").action("store").type("float").set_default("").help("(1,0) element of the tranformation matrix.");
+    parser.add_option("--m_11").action("store").type("float").set_default("").help("(1,1) element of the tranformation matrix.");
+    parser.add_option("--m_12").action("store").type("float").set_default("").help("(1,2) element of the tranformation matrix.");
+    parser.add_option("--m_20").action("store").type("float").set_default("").help("(2,0) element of the tranformation matrix.");
+    parser.add_option("--m_21").action("store").type("float").set_default("").help("(2,1) element of the tranformation matrix.");
+    parser.add_option("--m_22").action("store").type("float").set_default("").help("(2,2) element of the tranformation matrix.");
+    parser.add_option("--m_03").action("store").type("float").set_default("").help("(0,3) element of the tranformation matrix.");
+    parser.add_option("--m_13").action("store").type("float").set_default("").help("(1,3) element of the tranformation matrix.");
+    parser.add_option("--m_23").action("store").type("float").set_default("").help("(2,3) element of the tranformation matrix.");
```

Comment:
Having those --m_XX input arguments are too much. This tool should read the transformation file that has been written by the registration tool. use --transformationFile .